packages/common/src/common.ts

```diff
@@ -326,10 +375,10 @@ export class Common {
   /**
    * Internal helper for _buildParamsCache()
    */
-  protected _mergeWithParamsCache(params: HardforkConfig | EIPConfig) {
+  protected _mergeWithParamsCache(params: ParamsConfig) {
     this._paramsCache = {
       ...this._paramsCache,
-      ...params['params'],
+      ...params,
```

Comment:
Here also a possible shallow copy.

Reply:
Guess you mean "deep copy" but: yes!

Will all do in subsequent PR (not the state manager one)